pkg/networkservice/common/connect/server.go

```diff
@@ -84,14 +84,25 @@ func NewServer(clientFactory func(ctx context.Context, cc grpc.ClientConnInterfa
 func (c *connectServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
 	client, found := c.connectionIDToClientMap.Load(request.GetConnection().GetId())
 	if found {
-		return client.Request(ctx, request)
+		conn, err := client.Request(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+		request.Connection = conn
```

Comment:
I don't think we actually want to pass the connection through here... we want to at most pass through the ConnectionContext (and that may be subject to some filtering).

Reply:
I created #306 to track filtering and validation of connection context